### PR TITLE
Get tokens method for BaseLanguageModel, OpenAI, & ChatOpenAI

### PR DESCRIPTION
**Pull request by zachschillaci27**
# Get tokens method for BaseLanguageModel, OpenAI, & ChatOpenAI
Sometimes it's helpful to inspect the tokenization of your model to better understand how certain words are being tokenized. Therefore, I have split the preexisting `get_num_tokens` of `BaseLanguageModel` into a `get_tokens` method - which returns a list of integer tokens - and the same `get_num_tokens` method, now calling `get_tokens` under the hood. I have further implemented these methods for the `OpenAI` and `ChatOpenAI` language models.

## Before submitting

I updated the integration test here `tests/integration_tests/test_schema.py` to also check the tokenization of the default `GPT2TokenizerFast` tokenizer.

## Who can review?

Community members can review the PR once tests pass. Tag maintainers/contributors who might be interested:

@hwchase17, @agola11

### CODE REVIEW

File: langchain/base_language.py [hunk 1]
```diff
@@ -26,10 +26,7 @@ def _get_num_tokens_default_method(text: str) -> int:
     tokenizer = GPT2TokenizerFast.from_pretrained("gpt2")
 
     # tokenize the text using the GPT-2 tokenizer
-    tokenized_text = tokenizer.tokenize(text)
-
-    # calculate the number of tokens in the tokenized text
-    return len(tokenized_text)
+    return tokenizer.encode(text)
```
Comment (vowelparrot) [inline langchain/base_language.py:29]:
```
    return tokenizer.tokenize(text)
```

File: langchain/base_language.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 from langchain.schema import BaseMessage, LLMResult, PromptValue, get_buffer_string
 
 
-def _get_num_tokens_default_method(text: str) -> int:
+def _get_tokens_default_method(text: str) -> list[int]:
```
Comment (vowelparrot) [inline langchain/base_language.py:13]:
```suggestion
def _get_token_ids_default_method(text: str) -> list[int]:
```

File: langchain/base_language.py [hunk 3]
```diff
@@ -61,9 +58,13 @@ def predict_messages(
     ) -> BaseMessage:
         """Predict message from messages."""
 
+    def get_tokens(self, text: str) -> list[int]:
+        """Get the tokens present in the text."""
```
Comment (vowelparrot) [inline langchain/base_language.py:62]:
```suggestion
    """Get the tokens IDs from the text."""
```

File: langchain/chat_models/openai.py [hunk 4]
```diff
@@ -372,10 +372,11 @@ def get_num_tokens(self, text: str) -> int:
         enc = tiktoken.encoding_for_model(self.model_name)
 
         # encode the text using the GPT-3.5-Turbo encoder
-        tokenized_text = enc.encode(text)
+        return enc.encode(text)
 
-        # calculate the number of tokens in the encoded text
-        return len(tokenized_text)
+    def get_num_tokens(self, text: str) -> int:
```
Comment (vowelparrot) [inline langchain/chat_models/openai.py:377]: I think this would be unnecessary now since it would call the base class, right?

**Comment by zachschillaci27** [comment]
@vowelparrot Thanks for the comments. I realize I missed another change [here](https://github.com/zachschillaci27/langchain/blob/1b4ac91720a5e5ff93719b3dbf24351f08442177/langchain/llms/openai.py#L807), on the `OpenAIChat` class

**Pull request merged**